pkg/pusher/pusher.go

```diff
@@ -112,7 +112,7 @@ func (s *Service) chunksWorker(warmupTime time.Duration, tracer *tracing.Tracer)
 
 	// inflight.set handles the backpressure for the maximum amount of inflight chunks
 	// and duplicate handling.
-	chunks, repeat, unsubscribe := s.storer.SubscribePush(ctx, s.inflight.set)
+	chunks, _, unsubscribe := s.storer.SubscribePush(ctx, s.inflight.set)
```

Comment:
Is this change required?

Reply:
maybe we can discuss this further, but we considered a retry mechanism where instead of resetting the iterator we push the chunk again into the "feeding" channel `line:162`